sdk/PrebidDemandSDKLoaders/DemandSDKLoaders/FacebookAudienceNetwork/PBFacebookBannerAdLoader.m

```diff
@@ -0,0 +1,123 @@
+/*   Copyright 2017 APPNEXUS INC
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "PBFacebookBannerAdLoader.h"
+
+#import "PBConstants.h"
+
+struct FBAdSize {
+    CGSize size;
+};
+
+@implementation PBFacebookBannerAdLoader
+
+- (void)loadAd:(NSDictionary *)info {
+	NSString *bidPayload = (NSString *)info[@"adm"];
+	CGFloat height = [(NSString *)info[@"height"] floatValue];
+	CGSize adSize = CGSizeMake(-1, height);
+
+	// TODO nicole validate adSize against FBAdSize
+
+	// Load FBAdView using reflection so we can load the ad properly in the FBAudienceNetwork SDK
+	Class fbAdViewClass = NSClassFromString(@"FBAdView");
+	#pragma clang diagnostic push
+	#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+    if (fbAdViewClass != nil) {
+        SEL initMethodSel = NSSelectorFromString(@"initWithPlacementID:adSize:rootViewController:");
+        id fbAdViewObj = [fbAdViewClass alloc];
+        if ([fbAdViewObj respondsToSelector:initMethodSel]) {
+            NSMethodSignature *methSig = [fbAdViewObj methodSignatureForSelector:initMethodSel];
+            NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:methSig];
+
+            [invocation setSelector:initMethodSel];
+            [invocation setTarget:fbAdViewObj];
+
+            // Set arguments for init method
+            NSString *placementId = [self parsePlacementIdFromBidPayload:bidPayload];
+            struct FBAdSize fbAdSize;
```

Comment:
For this, there's no dependency on FB audience network framework?

Reply:
Here there isn't a dependency on the framework because above on line 20 I'm initializing an empty struct with the same name. I'm going to work on updating this to use reflection to pull in the constants like you do in your implementation.